sycl/test/intel_matrix/matrix-amx-bf16-test.cpp

```diff
@@ -0,0 +1,184 @@
+// RUN: %clangxx -mamx-bf16 -mamx-int8 -mavx512bw -mavx512vbmi -fsycl -DAMX -O2 %s
```

Comment:
```suggestion
// RUN: %clangxx -mamx-bf16 -mamx-int8 -mavx512bw -mavx512vbmi -fsycl -DAMX -O2 %s -o %t
```

I think we should execute this test, but I assume it requires HW support for AMX ISA. Right?
This might require additional LIT configuration:
1. Add new feature to the config, detect if HW supports it and enable it for LIT tests.
2. Test must require AMX feature to run.

According to https://github.com/intel/llvm/blob/sycl/CONTRIBUTING.md#tests-development, we should also move this test to https://github.com/intel/llvm/tree/sycl/sycl/test/on-device/ (https://github.com/intel/llvm/tree/sycl/sycl/test/on-device/extensions in particular) directory.

The same is applicable for the second test as well.

Reply:
Thanks for comments, Alexey. Eh, BTW, If the testcase can't run for now, could we still move it to sycl/test/on-device/extensions for now?

Reply:
If you are going to use it for testing "does it compile?" mode, than there is no need to move. 

The `on-device` directory is intended for the tests requiring special HW for execution and testing features under active development. When feature API is finalized and it's ready for end users, we move such tests to [llvm-test-suite](https://github.com/intel/llvm-test-suite/tree/intel/SYCL) repository.